firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -210,41 +210,42 @@ uint8_t moveDiagonal(int startCol, int startRow, int endCol, int endRow)
 
   digitalWrite(xMotor[DIR_PIN], dirX);
   digitalWrite(yMotor[DIR_PIN], dirY);
-  
+
   if (numStepsX == numStepsY)
   {
     setScale(xMotor, WHOLE_STEPS);
     setScale(yMotor, WHOLE_STEPS);
-    numEighthStepsX = 8;
-    numEighthStepsY = 8;
+    // Sets sign based off of direction of each motor
+    numEighthStepsX *= (dirX == RIGHT) ? 8 : -8;
+    numEighthStepsY *= (dirY == UP) ? 8 : -8;
   }
   else if (numStepsY > numStepsX && (numStepsY / numStepsX) == 2)
   {
     setScale(xMotor, HALF_STEPS);
     setScale(yMotor, WHOLE_STEPS);
-    numEighthStepsX = 4;
-    numEighthStepsY = 8;
+    // Sets sign based off of direction of each motor
+    numEighthStepsX *= (dirX == RIGHT) ? 4 : -4;
+    numEighthStepsY *= (dirY == UP) ? 8 : -8;
   }
   else if (numStepsY < numStepsX && (numStepsX / numStepsY) == 2)
   {
     setScale(xMotor, WHOLE_STEPS);
     setScale(yMotor, HALF_STEPS);
     numEighthStepsX = 8;
     numEighthStepsY = 4;
+    // Sets sign based off of direction of each motor
+    numEighthStepsX *= (dirX == RIGHT) ? 8 : -8;
```

Comment:
Same as above, but you also need to remove those assignments above this, they're unnecessary.